### PR TITLE
Misc fixes from model comp work

### PR DESCRIPTION
**Pull request by MattFiler**
### Added
- Added option to fade UIInputCursor sprite out based on specified interaction distance range
- Added range to customise the distance at which the UIInputCursor scales when using direct input

### Changed
- PostProcessProvider now defaults to UpdateAndFixedUpdate updating to better support PhysicalHandsManager out of the box
- Service providers using a specific serial will now connect to the exact given serial, rather than first closest
- UI PointerElement now only uses specified finger rather than the calculated closest to canvas (defaults to INDEX)
- Reduced the default UIInputCursor minimum sprite scale
- UIInputEventSystem prefab now defaults to Direct interaction mode
- Modified version check logic for readability

### Fixed
- Service providers using a specific serial will now correctly handle device reconnections
- Fixed instance where setting CapsuleHand colour through code would report a nullref
- Fixed references to LeapProvider and Camera in PointerElement to prevent nullrefs when changing either in the main UIInputModule
- Fixed GrabBall for new physical setup
- Correctly unassign/re-assign LeapProvider events in PhysicalHandsManager if reference changed at runtime
- Fixed an unreachable code warning if XR_MANAGEMENT and INPUT_SYSTEM are not AVAILABLE
- Fixed the default float height for UIInputCursor
- Increased timeout and retry delay in ServerStatus to reduce GetServerStatus fails
- Fixed an instance where ServerStatus would report a nullref if the devices array fails to populate
- Fixed MRTK integration (updated finger/bone methods for new 7.0.0 implementation)

## Contributor Tasks

- [ ] Create or edit test cases [here](https://ultrahaptics.atlassian.net/projects/UNITY?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/v2/testCases?projectId=15189)
- [ ] Add a CHANGELOG entry for this change.
- [ ] Ensure documentation requirements are met e.g., public API is commented.
- [ ] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.

## Test Cycle

_Link to the test cycle here._

## Reviewer Tasks

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [ ] All tests must be ran and cover all scenarios (If not, add new tests to the cycle and run them).
- [x] Documentation has been reviewed.
- [ ] Approve with a comment of any additional tests run or any observations.

## Related JIRA Issues

_If this MR closes any JIRA issues list them below in the form `Closes PROJECT-#`_

## Pull Request Templates

Switch template by going to preview and clicking the link - note it will not work if you've made any changes to the description.

- [default.md](?expand=1) - for contributions to stable packages.
- [release.md](?expand=1&template=release.md) - for release merge requests.

**You are currently using: default.md**

Note: these links work by overwriting query parameters of the current url. If the current url contains any you may want to amend the url with `&template=name.md` instead of using the link. See [query parameter docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request) for more information.

### CODE REVIEW

File: Packages/Tracking/Core/Editor/Scripts/LeapServiceProviderEditor.cs
Comment (MaxPalmer-UH) [inline Packages/Tracking/Core/Editor/Scripts/LeapServiceProviderEditor.cs:1]: Would prefer this to be kept in.

Comment (MattFiler) [reply Packages/Tracking/Core/Editor/Scripts/LeapServiceProviderEditor.cs:1]: Oops, yeah have re-enabled version check and actually fixed what was broken behind it 😅

**Comment by MaxPalmer-UH** [comment]
Just tried the UI input sample scene (ExampleUIScene), something is up with indirect interaction in this scene - initially the transition distance seemed to have been set incorrectly and then I couldn't change the button states using far field. I see there's been a few changes to the UI Input module so perhaps this is related?

**Comment by MaxPalmer-UH** [comment]
> Just tried the UI input sample scene (ExampleUIScene), something is up with indirect interaction in this scene - initially the transition distance seemed to have been set incorrectly and then I couldn't change the button states using far field. I see there's been a few changes to the UI Input module so perhaps this is related?

Sounds like this isn't a regression so good to go.

**Review by MaxPalmer-UH — APPROVED** [review]

**Pull request merged**